File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    map<string, int> m;
    for(int i=0; i<name.size(); i++) {
        m.insert({name[i], yearning[i]});
    }
    
    for(auto r : photo) {
        int ans=0;
        for(auto i : r) {
            if(m.find(i) != m.end()) ans += m.find(i)->second;
        }
        answer.push_back(ans);
    }
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    const map = {};
    for(let i=0; i<name.length; i++) {
        map[name[i]] = yearning[i]; 
    }

    for(p of photo) {
        let total = 0;
        for(let i=0; i<p.length; i++) {
            if(map[p[i]] !== undefined) {
                total += map[p[i]];   
            } 
        }
        answer.push(total);
    }
    return answer;
}